bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoBasicHandler.java

```diff
@@ -130,73 +132,94 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
         logger.debug("Locating action for channel {}: {}", channelUID.getId(), command);
         if (!actions.isEmpty()) {
-            if (actions.containsKey(channelUID.getId())) {
+            if (actions.containsKey(channelUID)) {
                 int valuePos = 0;
-                @Nullable
-                JsonElement value = null;
-                MiIoDeviceAction action = actions.get(channelUID.getId());
-                JsonArray parameters = actions.get(channelUID.getId()).getParameters();
-                for (int i = 0; i < action.getParameters().size(); i++) {
-                    JsonElement p = action.getParameters().get(i);
-                    if (p.isJsonPrimitive() && p.getAsString().toLowerCase().contains("$value$")) {
-                        valuePos = i;
+                MiIoBasicChannel miIoBasicChannel = actions.get(channelUID);
+                for (MiIoDeviceAction action : miIoBasicChannel.getActions()) {
+                    @Nullable
+                    JsonElement value = null;
+                    JsonArray parameters = action.getParameters();
+                    for (int i = 0; i < action.getParameters().size(); i++) {
+                        JsonElement p = action.getParameters().get(i);
+                        if (p.isJsonPrimitive() && p.getAsString().toLowerCase().contains("$value$")) {
+                            valuePos = i;
+                        }
                     }
-                }
-                String cmd = actions.get(channelUID.getId()).getCommand();
-                CommandParameterType paramType = actions.get(channelUID.getId()).getparameterType();
-                if (paramType == CommandParameterType.COLOR) {
-                    if (command instanceof HSBType) {
-                        HSBType hsb = (HSBType) command;
-                        Color color = Color.getHSBColor(hsb.getHue().floatValue() / 360,
-                                hsb.getSaturation().floatValue() / 100, hsb.getBrightness().floatValue() / 100);
-                        value = new JsonPrimitive(
-                                (color.getRed() * 65536) + (color.getGreen() * 256) + color.getBlue());
+                    String cmd = action.getCommand();
+                    CommandParameterType paramType = action.getparameterType();
+                    if (paramType == CommandParameterType.COLOR) {
+                        if (command instanceof HSBType) {
+                            HSBType hsb = (HSBType) command;
+                            Color color = Color.getHSBColor(hsb.getHue().floatValue() / 360,
+                                    hsb.getSaturation().floatValue() / 100, hsb.getBrightness().floatValue() / 100);
+                            value = new JsonPrimitive(
+                                    (color.getRed() * 65536) + (color.getGreen() * 256) + color.getBlue());
```

Comment:
This calculation would look better using bit-shift operators instead of multiplication.
```suggestion
                                    (color.getRed() << 16) + (color.getGreen() << 8) + color.getBlue());
```